libgdf/src/gdf_table.cuh

```diff
@@ -476,92 +477,32 @@ public:
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            my_elem = other_elem;
-            break;
-          }
-        case GDF_INT32:
-          {
-            using col_type = int32_t;
-            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            my_elem = other_elem;
-            break;
-          }
-        case GDF_INT64:
-          {
-            using col_type = int64_t;
-            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            my_elem = other_elem;
-            break;
-          }
-        case GDF_FLOAT32:
-          {
-            using col_type = float;
-            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            my_elem = other_elem;
-            break;
-          }
-        case GDF_FLOAT64:
-          {
-            using col_type = double;
-            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            my_elem = other_elem;
-            break;
-          }
-        case GDF_DATE32:
-          {
-            using col_type = int32_t;
-            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            my_elem = other_elem;
-            break;
-          }
-        case GDF_DATE64:
-          {
-            using col_type = int64_t;
-            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            my_elem = other_elem;
-            break;
-          }
-        case GDF_TIMESTAMP:
-          {
-            using col_type = int64_t;
-            col_type & my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
-            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
-            my_elem = other_elem;
-            break;
-          }
-        default:
-          return GDF_UNSUPPORTED_DTYPE;
-      }
+      gdf_type_dispatcher(my_col_type,
+                          copy_element,
+                          0,
+                          d_columns_data[i],
+                          my_row_index,
+                          other.d_columns_data[i],
+                          other_row_index);
+
```

Comment:
So, this is obviously better then before, but - I don't like seeing a call to a dispatcher function with a bunch of arguments. Are you absolutely sure we can't separate these two things? That is, have a higher-order function which returns a function-specific dispatcher? So that we could do:
```
auto copy_element = type_apply_and_erase(typed_copy_element, my_col_type);
copy_element(d_columns_data[i], my_row_index, other.d_columns_data[i], other_row_index);
```
I think this might be possible.

Reply:
I like this idea. I'd be curious if you have any ideas on how this could be implemented. I think it could be done with function pointers, but I'd like to using function pointers in device code.